src/Protocol/OStatus.php

```diff
@@ -1283,10 +1284,19 @@ private static function addHeader(DOMDocument $doc, array $owner, $filter)
 		$root->setAttribute("xmlns:mastodon", NAMESPACE_MASTODON);
 
 		$title = '';
+		$selfUri = '/feed/' . $owner["nick"] . '/';
```

Comment:
I would prefer having all ```$selfUri``` assignments at a single place.

Reply:
Like repeating this file for each of the switch cases?

Reply:
@annando New proposition https://github.com/friendica/friendica/pull/6144/commits/bd19e93c094b85bc2fb40044d972dcb98ce53d7c

Reply:
Why not simply:
```
if (!$feed_mode) {
	$selfUri = "/dfrn_poll/" . $owner["nick"];
} else {
	$selfUri** = '/feed/' . $owner["nick"] . '/';
}
```

Reply:
@annando Your code does not seem to work for for the 2 other types of feeds, does it?
```
/feed/nick/comments
/feed/nick/activity
```

Reply:
Ah, thanks!